sdks/java/harness/src/main/java/org/apache/beam/fn/harness/FnApiDoFnRunner.java

```diff
@@ -858,6 +1008,10 @@ public static WindowedSplitResult forRoots(
   private void processElementForWindowObservingSizedElementAndRestriction(
       WindowedValue<KV<KV<InputT, KV<RestrictionT, WatermarkEstimatorStateT>>, Double>> elem) {
     currentElement = elem.withValue(elem.getValue().getKey().getKey());
+    // No need to process if current restriction is null.
+    if (elem.getValue().getKey().getValue().getKey() == null) {
```

Comment:
I don't think this is the right way to go. The output received should produce 0 outputs instead of null.